### DISCUSSION TITLE
Question about Node v12 compatibility

### DISCUSSION THREAD

**Discussion by ozcanturkmen** [alexandercerutti/passkit-generator#112] [Q&A] [answered]
Hi, 

This will be a quick question rather than a feature request or a bug report.
In package.json the passkit-generator (v3.1.6) library has engines "node": ">=14.18.1" 
Though we had to use the it in a Node v12.16.3 runtime environment. 
No functional issues observed so far - besides getting the "Unsupported engine" NPM warning. 
I'd like to keep the latest version of the library whenever possible, because I love the changes. 
What is your suggestion on that? 
If you think I definitely must stick to v2 instead of v3 please let me know, because the solution is live in prod, and a Node upgrade is unfortunately not possible in the near future.

Thank you for your time, many thanks in advance!

**Comment by alexandercerutti** [comment, marked as answer]
Hey @ozcanturkmen, thanks for using passkit-generator!

Node v12 [is not even, anymore, in Maintainance Mode](https://github.com/nodejs/release#release-schedule). IMHO, you should do you best (and maybe prioritize if possible) to upgrade at least to v14, if not v16. 

Most of the libraries out there already support v14 and v16. On the next version I will release (I don't have any plans yet actually), I'll bump it to the next LTS version supported, so you risk to always remain behind on this situation.

Besides this platform version discussion, if you actually didn't detect any issue, I think the warning can be safely ignored. Maybe it might worth documenting the decision until you won't upgrade to, at least, Node v14 (but please note that v14 will get out of maintainance mode in April 2023).

Otherwise you can downgrade to v2.0.8, which is the latest v2.x that received most important changes before the v3.x almost complete rewrite (schema fixes and removal of moment-js).

I think the best solution for you and the viable one is the first one, so your company doesn't have to waste man-hour time to perform a rollback on this library.

======

Just as a side question, you said that you love v3's changes. May I ask you which are the changes you love the most or are more useful to you? Just as a feedback, you know 😄

**Reply by ozcanturkmen** [reply]
@alexandercerutti Thank you for the valuable information you provided! Just to answer your side question: I like _addBuffer_ and _props_ in v3 :)
